### DISCUSSION TITLE
How to do device memory profiling in Colab for JAX/Flax?

### DISCUSSION THREAD

**Discussion by marcvanzee** [google/flax#918] [Q&A] [answered]

**Comment by marcvanzee** [comment, marked as answer]
Some simple tips:
- [JAX memory profiling](https://jax.readthedocs.io/en/latest/device_memory_profiling.html) can provide help to spot where the issue occurs.
- you could run the cell with `%%jaxheapprof`  to get the device memory profile from Colab.

**Reply by ayaka14732** [reply]
> you could run the cell with %%jaxheapprof to get the device memory profile from Colab.

This gives an error on Colab

![](https://user-images.githubusercontent.com/68557794/154809681-87a1cd5b-5a1f-4332-9728-6069d2a9aecf.png)
